tester.go

```diff
@@ -18,7 +20,20 @@ type Tester struct {
 
 // newTester creates a Tester based on the TesterDefinition provided
 func newTester(env map[string]string, definition tester_definition.TesterDefinition) (Tester, error) {
-	context, err := tester_context.GetTesterContext(env, definition.ExecutableFileName)
+	submissionDir, ok := env["CODECRAFTERS_SUBMISSION_DIR"]
```

Comment:
This assertion is currently present inside `GetTesterContext`. Maybe we should just pass in `definition` to `GetTesterContext`, and perform the rest there? Seems weird to duplicate this check here

Reply:
Yeah, agreed! Will move this there